sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -296,32 +299,42 @@ private Optional<SclReportItem> updateSourceDataSetsAndControlBlocks(TExtRef ext
             return warningReportItem(extRef, String.format(MESSAGE_SOURCE_LN_NOT_FOUND, optionalSourceLDevice.get().getXPath()));
         }
 
-        //TODO: Don't forget to test this in issue #84 RSR-433
-        List<ResumedDataTemplate> mxAndStSources = sourceDas.stream()
-            .filter(da -> da.getFc() == TFCEnum.MX || da.getFc() == TFCEnum.ST).toList();
+        return filterValidSourceDas(extRef, sourceDas);
+    }
 
-        Stream<ResumedDataTemplate> sourceDaToProcess;
+    private Optional<SclReportItem> filterValidSourceDas(TExtRef extRef, final Set<ResumedDataTemplate> sourceDas) {
+        //TODO: Don't forget to test this in issue #84 RSR-433
+        sourceDas.removeIf(da -> da.getFc() != TFCEnum.MX && da.getFc() != TFCEnum.ST);
```

Comment:
Are this condition exactly same as  .filter(da -> da.getFc() == TFCEnum.MX || da.getFc() == TFCEnum.ST).toList(); ?
What should happen when da.getFC()==null ?

Reply:
Yes it it is the same.
When `da.getFC()==null` : 
* .filter `null == TFCEnum.MX || null == TFCEnum.ST` => false => not in the collected list
* .removeIf `null != TFCEnum.MX && null != TFCEnum.ST)` => true => removed from list

Reply:
ok

Reply:
filterValidSourceDas could be refactor